clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7115,6 +7115,9 @@ def warn_shift_result_sets_sign_bit : Warning<
   "signed shift result (%0) sets the sign bit of the shift expression's "
   "type (%1) and becomes negative">,
   InGroup<DiagGroup<"shift-sign-overflow">>, DefaultIgnore;
+def warn_shift_bool : Warning<
+  "%select{left|right}0 shifting a `bool` implicitly converts it to 'int'">,
```

Comment:
I wonder if we should also suggest to change the shift to `b & !x` for right shifts?

Reply:
@Fznamznon we can make a suggestion, however, I'm not confident about one for the right shift.